lib/spack/spack/compilers/clang.py

```diff
@@ -29,22 +48,29 @@ class Clang(Compiler):
     # Subclasses use possible names of Fortran 90 compiler
     fc_names = ['flang', 'gfortran', 'xlf90_r']
 
-    # Named wrapper links within lib/spack/env
-    link_paths = {'cc': 'clang/clang',
-                  'cxx': 'clang/clang++'}
-
-    if sys.platform == 'darwin':
-        # Use default wrappers for fortran, in case provided in
-        # compilers.yaml
-        link_paths['f77'] = 'clang/gfortran'
-        link_paths['fc'] = 'clang/gfortran'
-    elif spack.architecture.sys_type() == 'linux-rhel7-ppc64le':
-        # This platform uses clang with IBM XL Fortran compiler
-        link_paths['f77'] = 'xl_r/xlf_r'
-        link_paths['fc'] = 'xl_r/xlf90_r'
-    else:
-        link_paths['f77'] = 'clang/flang'
-        link_paths['fc'] = 'clang/flang'
+    # Clang has support for using different fortran compilers with the
+    # clang executable.
+    @property
+    def link_paths(self):
+        # clang links are always the same
+        link_paths = {'cc': 'clang/clang',
+                      'cxx': 'clang/clang++'}
+
+        # fortran links need to look at the actual compiler names from
+        # compilers.yaml to figure out which named symlink to use
+        for string, link_path in f77_mapping:
+            if string in self.f77:
```

Comment:
@tgamblin 

A more descriptive variable name is preferable: `compiler_name`?

also, `self.f77` is a path. It is slightly confusing to use `if string in self.f77`, would `if self.f77 and self.f77.endswith(string)` work?

Reply:
@scheibelp: I used `in` because there are lots of ways that compiler can be named.  For example, some sites deploy with architecture prefixes and version suffixes:
 
```
linux-unknown-x86_64-gfortran-5.5.0
```

Note that we also detect names like this in Spack (look at `gcc.py`)

So, `endswith` is not sufficient.  I want users to be able to put whatever their compiler is called into `compilers.yaml`.

Reply:
I updated the name to `compiler_name` -- it's accurate.